Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift

```diff
@@ -1,63 +1,70 @@
+        super.init(
+            gestureRecognizer: gestureRecognizer,
+            mapboxMap: mapboxMap,
+            cameraAnimationsManager: cameraAnimationsManager)
+        gestureRecognizer.addTarget(self, action: #selector(handleGesture(_:)))
     }
 
-    @objc internal func handlePinch(_ pinchGestureRecognizer: UIPinchGestureRecognizer) {
-
-        delegate.cancelGestureTransitions()
-        let pinchCenterPoint = pinchGestureRecognizer.location(in: view)
-
-        if pinchGestureRecognizer.state == .began {
-
-            self.previousScale = 1.0
-            delegate.gestureBegan(for: .pinch)
+    @objc private func handleGesture(_ gestureRecognizer: UIPinchGestureRecognizer) {
+        guard let view = gestureRecognizer.view else {
+            return
+        }
 
-            self.initialCameraState = self.delegate.cameraState()
-            self.initialPinchCenterPoint = pinchCenterPoint
+        let pinchCenterPoint = gestureRecognizer.location(in: view)
 
-            /**
-             TODO: Handle a concurrent rotate gesture here.
-             Prioritize the correct gesture by comparing the velocity of competing gestures.
-             */
-        } else if pinchGestureRecognizer.state == .changed {
-            if pinchGestureRecognizer.numberOfTouches < 2 {
+        switch gestureRecognizer.state {
+        case .began:
+            initialPinchMidpoint = pinchCenterPoint
+            initialCameraState = mapboxMap.cameraState
+            cameraAnimationsManager.cancelAnimations()
+            delegate?.gestureBegan(for: .pinch)
+        case .changed:
+            guard gestureRecognizer.numberOfTouches == 2,
+                  let initialCameraState = initialCameraState,
+                  let initialPinchCenterPoint = initialPinchMidpoint else {
                 return
             }
+            cameraAnimationsManager.cancelAnimations()
+
+            let zoomIncrement = log2(gestureRecognizer.scale)
+            var cameraOptions = CameraOptions()
+            cameraOptions.center = initialCameraState.center
+            cameraOptions.padding = initialCameraState.padding
+            cameraOptions.zoom = initialCameraState.zoom
 
-            let zoomIncrement = log2(pinchGestureRecognizer.scale)
-            delegate.pinchChanged(withZoomIncrement: zoomIncrement,
-                                  targetAnchor: pinchCenterPoint,
-                                  initialAnchor: self.initialPinchCenterPoint,
-                                  initialCameraState: self.initialCameraState)
+            mapboxMap.setCamera(to: cameraOptions)
 
-            previousScale = pinchGestureRecognizer.scale
-        } else if pinchGestureRecognizer.state == .ended
-            || pinchGestureRecognizer.state == .cancelled {
+            mapboxMap.dragStart(for: initialPinchCenterPoint)
+            let dragOptions = mapboxMap.dragCameraOptions(
+                from: initialPinchCenterPoint,
```

Comment:
nit: indentation feels weird here

Reply:
See https://github.com/mapbox/mapbox-maps-ios/pull/677#discussion_r709487225

Reply:
Resolving. Let's centralize the conversation in the linked thread.